homeassistant/components/geniushub/config_flow.py

```diff
@@ -0,0 +1,179 @@
+
+from __future__ import annotations
+
+from http import HTTPStatus
+import logging
+import socket
+from typing import Any
+
+import aiohttp
+
+from homeassistant import config_entries
+from homeassistant.config_entries import ConfigFlowResult
+
+from . import V1_API_SCHEMA, V3_API_SCHEMA, validate_input
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class GeniusHubConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for Solis logger."""
+
+    VERSION = 1
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> ConfigFlowResult:
+        """User config step for determine v1 or v3."""
+        _LOGGER.debug("ConfigFlow.async_step_user")
+        return self.async_show_menu(
+            step_id="user",
+            menu_options=["v3_api", "v1_api"],
+        )
+
+    async def async_step_v3_api(
+        self, user_input: dict[str, Any] | None = None
+    ) -> ConfigFlowResult:
+        """Version 3 configuration."""
+        _LOGGER.debug("ConfigFlow.async_step_v3_api: ", extra=user_input)
+        if user_input is None:
+            return self.async_show_form(step_id="v3_api", data_schema=V3_API_SCHEMA)
+
+        self._async_abort_entries_match(user_input)
+        errors = {}
+
+        try:
+            info = await validate_input(self.hass, user_input)
+
+        except socket.gaierror:
+            errors["base"] = "invalid_host"
+
+        except aiohttp.ClientResponseError as err:
+            if err.status == HTTPStatus.UNAUTHORIZED:
+                errors["base"] = "unauthorized"
+            else:
+                errors["base"] = "invalid_host"
+
+        except TimeoutError:
+            errors["base"] = "cannot_connect"
+
+        except aiohttp.ClientConnectionError:
+            errors["base"] = "cannot_connect"
+
+        except Exception as e:  # pylint: disable=broad-except
+            _LOGGER.error("Error in genius hub client", exc_info=e)
+            _LOGGER.exception("Unexpected exception")
+            errors["base"] = "unknown"
+        else:
+            _LOGGER.debug(
+                "ConfigFlow.async_step_v3_api: validation passed:",
+                extra=user_input,
+            )
+            return self.async_create_entry(title=info["title"], data=user_input)
+
+        _LOGGER.debug(
+            "ConfigFlow.async_step_v3_api: validation failed: ",
+            extra=user_input,
+        )
+
+        return self.async_show_form(
+            step_id="user",
+            errors=errors,
+        )
+
+    async def async_step_v1_api(
+        self, user_input: dict[str, Any] | None = None
+    ) -> ConfigFlowResult:
+        """Version 1 configuration."""
+        _LOGGER.debug("ConfigFlow.async_step_v1_api: ", extra=user_input)
+        if user_input is None:
+            return self.async_show_form(step_id="v1_api", data_schema=V1_API_SCHEMA)
+
+        self._async_abort_entries_match(user_input)
```

Comment:
We can check for the mac address only. I imagine there aren't 2 tokens, so if it's already registered, we can just abort if we see the same macaddress

Reply:
Not sure. You need the user name and password to obtain the mac address.

Reply:
yes, but in those cases, can we get the mac address from the service in the config flow

Reply:
The local service (V3) returns the MAC address.

The cloud service (V1) requires the MAC address as a key that is used in HA. In reality this can be any unique identifier. The actual service that is called does not need the MAC address or can provide it.

Reply:
Can it return something else unique?

Reply:
As in, we need something unique from the device without the user having to put it in. Otherwise they can make a typo and setup the device twice and that's a bad user experience so we want to avoid that.

If we can't do that, we should compare the user input with what we know to avoid duplicates. But that means we probably have to re review the unique IDs of the entities as well

Reply:
For local the UID is the MAC address.
For Cloud - I can find no UID. 

For the 25 users that use this integration I feel this is good enough and I am now wasting my time.

I am running this locally and will probably abandon this PR.

Reply:
Would be sad to see, I think we are close to getting this ready.

Reply:
I see nothing to change here. There is no unique ID other than what the user enters or the MAC address.